src/provider/common/Signal.ts

```diff
@@ -0,0 +1,262 @@
+// tslint:disable-next-line:no-any
```

Comment:
Yes, should definitely do that at some point - but not sure yet how best to do that.

A single-file `openfin-services-signal` module would be way too granular, but not sure right now what else it would make sense to group this with. Once the provider is built-out a bit more, we can make a call on how to modularise it.

Reply:
Fair enough. Wasn't expecting it to be in this PR, but wanted to start the conversation.

Reply:
It's not even particularly services-specific (or even openfin-specific), so long-term could just split it out as a standalone 'simple-signals' module. Either way, discussion for later